src/common/components/repository-input/index.js

```diff
@@ -0,0 +1,70 @@
+import 'isomorphic-fetch';
+
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import {
+  changeRepositoryInput,
+  verifyGitHubRepository
+} from '../../actions/repository-input';
+
+export class RepositoryInput extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  getStatusMessage() {
+    const input = this.props.repositoryInput;
+    let message;
+
+    if (!input.repository) {
+      message = 'Invalid repository URL.';
+    } else if (input.repository && input.isFetching) {
+      message = `Verifying ${input.repository} on GitHub...`;
+    } else if (input.success && input.repositoryUrl) {
+      message = `Repository ${input.repository} is valid.`;
+    } else if (input.errors) {
+      message = `Repository ${input.repository} is invalid.`;
+    }
+
+    return message;
+  }
+
+  render() {
+    return (
+      <div>
+        <label>Repository URL:</label>
+        <input type='text' value={this.props.repositoryInput.inputValue} onChange={this.onInputChange.bind(this)} />
+        <button onClick={this.onButtonClick.bind(this)}>Parse</button>
+        <div>
+          {this.getStatusMessage()}
+        </div>
+      </div>
+    );
+  }
+
+  onInputChange(event) {
+    this.props.dispatch(changeRepositoryInput(event.target.value));
+  }
+
+  onButtonClick() {
+    this.props.dispatch(verifyGitHubRepository(this.props.repositoryInput.inputValue));
+  }
+}
+
+RepositoryInput.propTypes = {
+  repositoryInput: PropTypes.object.isRequired,
+  dispatch: PropTypes.func.isRequired
+};
+
+function mapStateToProps(state) {
```

Comment:
@bartaz: Maybe "container" wasn't the right word in this case. I think presentational components (such as RepositoryInput) should contain logic only for changing their appearance and defer responding to user input to their parent component. [This article](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.c0azv4409) sums up what I mean ("presentational components have no dependencies on the rest of the app, such as Flux actions or stores").

In this case, I think it would be better to pass functions in as props like we do with the [InputField component in JavanRhino](https://github.com/canonical-ols/javan-rhino/blob/master/src/common/components/forms/inputField.js#L52).

I think there's a value in all our form components behaving in similar ways and having similar interfaces. It would also allow us to have two instances of RepositoryInput on the same page, which is an unlikely situation but could prevent some tricky bugs.